app/src/generator/Radii/Radii.tsx

```diff
@@ -16,19 +16,57 @@ const Radii = () => {
   const [currValues, setCurrValues] = useState<Map<string, string | number>>(
     new Map<string, string | number>()
   );
+  const currValuesRef = useRef<Map<string, string | number>>(currValues);
 
-  const valueChangedHandler = (spacing: string, value: string) => {
+  const valueChangedHandler = (radii: string, value: string) => {
     const map = new Map<string, string | number>(currValues);
-    map.set(spacing, value);
+
+    const storedValue =
+      currValuesRef.current.get(radii) ||
+      customTheme.radii[radii as keyof HvThemeTokens["radii"]];
+
+    const unit = extractFontSizeUnit(storedValue.toString());
+
+    map.set(radii, `${value}${unit}`);
+    currValuesRef.current = map;
     setCurrValues(map);
   };
 
   const setValueHandler = (radii: string) => {
-    const radiiValue = currValues.get(radii) || 0;
+    const value = currValuesRef.current.get(radii);
+    if (!value) return;
+
+    const radiiValue = parseInt(value.toString(), 10) || 0;
```

Comment:
```suggestion
    const radiiValue = Number(value) || 0;
```
`Number` is more modern - no need for `radix` and accept any type